code/game/atoms_movable.dm

```diff
@@ -713,6 +714,16 @@
 		else if(new_turf && !old_turf)
 			SSspatial_grid.enter_cell(src, new_turf)
 
+	// Z-Mimic hook
+	if (bound_overlay)
+		// The overlay will handle cleaning itself up on non-openspace turfs.
+		if (new_turf)
+			bound_overlay.forceMove(get_step(src, UP))
```

Comment:
is raw get_step() safe like this?

Reply:
It's safe, but only because of ZM's linearity requirement/assumption.